src/lib.rs

```diff
@@ -1,287 +1,340 @@
+                    stdout_lock.flush().unwrap();
+                    drop(stdout_lock);
+
+                    thread::sleep(std::time::Duration::from_millis(
+                        spinner_data.interval as u64,
+                    ));
+                }
+
+                delete_last_line(last_length);
+            }
+        });
+
+        // Return a Spinner struct so we can implement methods on it instead of `spinoff::stop()` etc.
+        Self {
+            thread_handle: Some(handle),
+            still_spinning,
+            msg,
+        }
+    }
+
     /// Stop the spinner.
     ///
     /// # Example
-    /// ```rust
-    /// use spinoff::Spinners;
-    /// use std::thread::sleep;
-    /// use std::time::Duration;
     ///
-    /// let sp = spinoff::new(Spinners::Dots, "Hello", None);
+    /// ```
+    /// # use spinoff::{Spinners, Spinner};
+    /// # use std::thread::sleep;
+    /// # use std::time::Duration;
+    /// #
+    /// let sp = Spinner::new(Spinners::Dots, "Hello", None);
     /// sleep(Duration::from_millis(800));
     /// sp.stop();
     /// ```
+    ///
     /// # Notes
-    /// * The spinner will be deleted after this method is called, the message will remain though.
-    /// * This method also sets the `still_spinning` atomic bool to `false`, which stops the spinner thread.
-    /// * This method cannot be called if the spinner is already stopped.
     ///
-    pub fn stop(self) {
-        self.still_spinning
-            .store(false, std::sync::atomic::Ordering::Relaxed);
-        print!("\r");
-        println!("{}    ", self.msg);
+    /// * The spinner will be dropped after this method is called, the message will remain though.
+    ///
+    pub fn stop(mut self) {
+        self.stop_spinner_thread();
+
+        // print message
+        println!("{}", self.msg);
     }
 
     /// Stops the spinner and prints a message on a new line.
     ///     
     /// # Example
-    /// ```rust
-    /// use spinoff::Spinners;
-    /// use std::thread::sleep;
-    /// use std::time::Duration;
     ///
-    /// let sp = spinoff::new(Spinners::Dots, "Hello", None);
+    /// ```
+    /// # use spinoff::{Spinners, Spinner};
+    /// # use std::thread::sleep;
+    /// # use std::time::Duration;
+    /// #
+    /// let sp = Spinner::new(Spinners::Dots, "Hello", None);
     /// sleep(Duration::from_millis(800));
     /// sp.stop_with_message("Bye");
     /// ```
     ///
-    /// # Panics
-    /// * The method will panic if the color is not one of the following: `blue`, `green`, `red`, `yellow`, `cyan`, `white` or `None`.
-    ///
-    /// # Notes
-    /// * This method cannot be called if the spinner is already stopped.
-    ///
-    pub fn stop_with_message(self, msg: StringLiteral) {
-        self.stop();
-        println!("{}", &msg);
+    pub fn stop_with_message(mut self, msg: &str) {
+        self.stop_spinner_thread();
+
+        // put the message over the spinner
```

Comment:
Originally, the `stop_with_message` function stopped the spinner (exactly like your `stop` implementation does), and printed the message on a new line, not replacing the message from the `stop` method.

Reply:
This could be fixed by moving the `delete_last_line` function call (the one after the thread stops) into `stop_spinner_thread` instead.